app/controllers/redirections_controller.rb

```diff
@@ -1,4 +1,6 @@
 class RedirectionsController < ApplicationController
+  before_action :ensure_referrer_is_not_localhost
```

Comment:
@edwardloveall, do you mean something like this code?

```ruby
class RedirectionsController
  rescue_from ValidationError, with: :redirect_to_localhost

  def next
    Validator.new(referrer).validate! # ValidationError comes from here

    redirection = find_or_create_redirection

    redirect_to redirection.next_url
  end

  private

  def redirect_to_localhost
    # ...
  end
end
```

I find `rescue_from` hard to reason about:

* When reading the `next` method, it's not clear that `Validator.new.validate!` can trigger a jump to a different method, because there's no `rescue` in that method.
* When reading the `rescue_from`, I personally can't tell where the `ValidationError` might come from (it could even be triggered by a method in `ApplicationController`). I must reason about the whole class to figure out what triggers the error and where it jumps from.
* `rescue_from` is using exceptions as control flow. Here's [a pretty good summary][so] of some arguments against exceptions as control flow. And [here's another][c2].

[so]: http://stackoverflow.com/a/729404
[c2]: http://c2.com/cgi/wiki?DontUseExceptionsForFlowControl

I don't have the same problems with `before_action`:

* `before_action`s are executed at a well-defined time (before each request)
* They cannot control the execution flow once a request "gets past" them -- if a controller method like `next` is being executed, the `before_action` can have no further effect.

That said, I'd be open to a plain `if` statement instead of the `before_action`:

```ruby
def next
  if referrer_is_localhost?
    redirect_to page_path("localhost")
  else
    # current implementation
  end
end
```